evals/api.py

```diff
@@ -23,85 +24,139 @@
 logger = logging.getLogger(__name__)
 
 
-def completion_query(
-    model_spec: ModelSpec,
-    prompt: Union[OpenAICreatePrompt, OpenAICreateChatPrompt, Prompt],
-    **kwargs,
-) -> tuple[dict, Union[OpenAICreatePrompt, OpenAICreateChatPrompt], dict]:
-    """
-    Query the API for a completion.
+class ReturnType(ABC):
+    def __init__(self, raw_data: Any, prompt: Any, metadata: Dict[str, Any] = None):
+        self.raw_data = raw_data
+        self.prompt: str = prompt
+        self.completions: List[str] = self.extract_completions()
+        self.metadata = metadata if metadata is not None else {}
```

Comment:
What do you think about having this base class simply consist of `extract_completions` and leave it up to the child classes to add their own variables as needed? I imagine some child classes might not require the same set or want to use the same naming.